cmd/neofs-node/config.go

```diff
@@ -559,7 +559,7 @@ func initCfg(appCfg *config.Config) *cfg {
 	c.internals = internals{
 		ctx:          context.Background(),
 		appCfg:       appCfg,
-		internalErr:  make(chan error),
+		internalErr:  make(chan error, 10), // We only need one error, but we can have multiple senders.
```

Comment:
looks like a linear retraction of the problem to me, although I believe that in practice it is now quite working

if we really
> need one error

maybe we'll have channel with capacity = 1 and write errors w/o blocking?
```go
select {
  case internalErr <- err:
  default:
    // return or smth else
}
```

btw who closes the channel?

Reply:
https://github.com/nspcc-dev/neofs-node/blob/6f7127a9ea141b8cfd490c8cdc5284cf27dbff73/cmd/neofs-node/main.go#L133-L134
And that's a problem at the moment. The other approach is leaving some drain routine, but that won't exit as well.

Reply:
ah, reader closes the channel, always not so good

 idk, drop closing and non-blocking writing LGTM